distutils/command/install.py

```diff
@@ -515,9 +520,17 @@ def finalize_other(self):
                       "I don't know how to install stuff on '%s'" % os.name)
 
     def select_scheme(self, name):
+        os_name, sep, key = name.partition('_')
```

Comment:
I acknowledge this approach is a little janky. I'd prefer to change the signature to accept the key instead, but I want to maintain the existing interface in case some of the transitional implementations are overriding it.